### PR TITLE
examples: sync bucket uploader example library

### PR DESCRIPTION
**Pull request by liavt**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            object_key = file_name
+                                                
+        self._update_object_status(object_key, zones)
+        
+        if timeout >= 0:
+            timer = threading.Timer(timeout, lambda: self._cancel(object_key))
+            timer.start()
+            
+        # Put objects to the relevant bucket
+        ans = self._s3_clients[zone_name].upload_file(Filename=file_name, Bucket=self._bucket_name,
```
Comment (yuvalif) [inline examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py:114]: probably good to wrap with a "try-expect" on "key exception" for the case that the zone provided didn't have a client

File: examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        # Create new Kafka consumer to listen to the message from Ceph
+        consumer = KafkaConsumer(
+            topic_name,
+            bootstrap_servers=kafka_endpoint,
+            value_deserializer=lambda x: loads(x.decode("utf-8")))
+    
+        for msg in consumer:
+            message = msg.value
+            if message['s3']['bucket']['name'] == self._bucket_name and message["s3"]['object']['key'] in self._objects \
+                    and message['eventName'] == "ceph:ObjectSynced":
```
Comment (yuvalif) [inline examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py:62]: the event name should be just `ObjectSynced`

**Comment by yuvalif** [comment]
@liavt please sign the commit and take out of draft mode

**Comment by cbodley** [comment]
> * would be nice to add a markdown file in that directory, showing how to use the library, and setup the environment for it to work (e.g. kafka setup)

i think this part is important

**Comment by yuvalif** [comment]
@omert please note that in the last commit you uploaded unmerged version of the code.
also, you need to sign your commit. use `git commit -s`

File: examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        self._s3_clients = {}
+        self._sns_clients = {}
+        
+        self._bucket_name = bucket_name
+        
+        topic_name = base64.b16encode(("Sync-" + bucket_name + "-" + kafka_endpoint).encode()).decode("utf-8")
+        
+        for zone in rgw_endpoints:
+            self._s3_clients[zone] = boto3.client('s3',
+                              endpoint_url=rgw_endpoints[zone],
```
Comment (yuvalif) [inline examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py:21]:
```suggestion
                              endpoint_url=zone,
```

File: examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python
```
Comment (yuvalif) [inline examples/boto3/sync_bucket_uploader_library/sync_bucket_uploader.py:1]: this is a module not a script, so you can remove this line

**Pull request closed without merging**